### PR TITLE
Extract shardProwConfig and refactor it to re-use in tide shard branch cut tool

### PR DESCRIPTION
**Pull request by jmguzik**
Signed-off-by: Jakub Guzik <jguzik@redhat.com>

### CODE REVIEW

**Comment by jmguzik** [comment]
/uncc @smg247 @bbguimaraes 
/cc @danilo-gemoli @droslean 

I think I can re-use this function in my tool by injecting additional operations in two places.

File: pkg/api/shardprowconfig/shardprowconfig.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	"k8s.io/test-infra/prow/github"
+
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/prowconfigsharding"
+)
+
+type QueryModifier func(*prowconfig.TideQuery, string)
+type ProwConfigGetter func(*prowconfig.ProwConfig)
+
+type ShardProwConfigFunctors struct {
```
Comment (droslean) [inline pkg/api/shardprowconfig/shardprowconfig.go:22]: This can be an interface instead.

**Pull request merged**